ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1049,6 +1061,10 @@ class PutChunk {
     // Tracks quota charging for this chunk.
     private OperationQuotaCharger operationQuotaCharger;
 
+    // Whether this chunk is compressed.  Default is false for not compressed.
+    // This value is set after compression has completed.  It is used to create PutRequest.
+    private boolean isChunkCompressed;
```

Comment:
please reset this variable back to false in clear method.

Reply:
ok thank you.